src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceResultSet.java

```diff
@@ -1173,11 +1193,18 @@ private String getColumnNameByIndex(int columnIndex) throws SQLException {
 
     private void getMoreData() throws SQLException {
         log.trace("Fetching page with number {} for resultset {}", ++currentPageNum, this);
-        ResultSet resultSet = statement.getNextPage();
+        ResultSet resultSet;
+
+        if(isPrestoPaginationRequired()) {
+            resultSet = statement.getNextPageFromBatchId(statement.nextBatchId);
```

Comment:
i would hold the nextBatchId as part of result set and not part of the statement.